src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -132,29 +143,36 @@ void register_UserAgeLessThenEighteen_notOk() {
     @DisplayName("register_throws_UserRegistrationException_"
             + "if_user_already_exists")
     void register_UserAlreadyExists_notOk() {
-        User oldUser = new User(1L, "valid login", "valid password", 18);
-        storageDao.add(oldUser);
-        newUser = new User(2L, "valid login", "other valid password", 18);
+        storageDao.add(validUser);
+        User invalidUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
         expectedMessage = "User with login %s already exists";
         UserRegistrationException expectedException = assertThrows(
                 UserRegistrationException.class,
-                () -> registrationService.register(newUser)
+                () -> registrationService.register(invalidUser)
         );
         assertEquals(
-                String.format(expectedMessage, newUser.getLogin()),
+                String.format(expectedMessage, invalidUser.getLogin()),
                 expectedException.getMessage()
         );
     }
 
     @Test
     @DisplayName("register_returns_new_registered_user")
     void register_registerNewUser_ok() {
-        User expectedUser = new User(1L, "test login", "test password", 18);
-        User actualUser = new User(1L, "test login", "test password", 18);
+        User expectedUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        expectedUser.setId(1L);
         int expectedSize = Storage.people.size() + 1;
-        assertAll(
-                () -> assertEquals(expectedUser, registrationService.register(actualUser)),
-                () -> assertEquals(expectedSize, Storage.people.size())
+        assertNull(
+                validUser.getId(),
+                "User id must be null before registration complete"
+        );
+        assertEquals(
+                expectedUser,
+                registrationService.register(validUser)
+        );
+        assertEquals(
+                expectedSize,
+                Storage.people.size()
```

Comment:
this can work too, but we could have been more precise here and check whether it is actually the same user that we have registered instead of just size check